clang/lib/Driver/Driver.cpp

```diff
@@ -2002,6 +2002,44 @@ void Driver::PrintVersion(const Compilation &C, raw_ostream &OS) const {
   // Print out the install directory.
   OS << "InstalledDir: " << Dir << '\n';
 
+  // Print out build configuration options that impact the compiler's runtime
+  // behavior. Intended for identifying the source of issues when reproducing
+  // changes.
+  std::vector<std::string> BuildOptions = {
+#if !__OPTIMIZE__
+      "+unoptimized",
+#endif
+#ifndef NDEBUG
+      "+assertions",
+#endif
+#ifdef EXPENSIVE_CHECKS
+      "+expensive-checks",
+#endif
+#if __has_feature(address_sanitizer)
+      "+asan",
+#endif
+#if __has_feature(undefined_behavior_sanitizer)
+      "+ubsan",
+#endif
+#if __has_feature(memory_sanitizer)
+      "+msan",
+#endif
+#if __has_feature(dataflow_sanitizer)
+      "+dfsan",
+#endif
+  };
+  if (!BuildOptions.empty()) {
```

Comment:
Do we want it to be hidden for the default config?
My opinion: yes—because this is helpful to llvm developers but not to compiler end-users.

Reply:
Making it opt-out, and having the release scripts disable it seems like a good idea to me. It's not actionable for end-users, but if one has to remember to turn it on (or even know it's an option), one might not see the benefit of it in day-to-day development.

Reply:
Added this as `LLVM_VERSION_PRINTER_SHOW_BUILD_CONFIG` which defaults on, is there anything I should do to update the release scripts here or is that for someone else to do?